x-pack/plugins/file_upload/public/components/geojson_upload_form/index_name_form.tsx

```diff
@@ -5,23 +5,46 @@
  * 2.0.
  */
 
+import _ from 'lodash';
 import React, { ChangeEvent, Component } from 'react';
 import { EuiFormRow, EuiFieldText, EuiCallOut, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { validateIndexName } from '../../util/indexing_service';
+import { validateIndexName } from '../../validate_index_name';
 
 export interface Props {
   indexName: string;
   indexNameError?: string;
   onIndexNameChange: (name: string, error?: string) => void;
+  onIndexNameValidationStart: () => void;
+  onIndexNameValidationEnd: () => void;
 }
 
 export class IndexNameForm extends Component<Props> {
-  _onIndexNameChange = async (event: ChangeEvent<HTMLInputElement>) => {
+  private _isMounted = false;
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  _onIndexNameChange = (event: ChangeEvent<HTMLInputElement>) => {
     const indexName = event.target.value;
+    this.props.onIndexNameChange(indexName);
+    this._validateIndexName(indexName);
+    this.props.onIndexNameValidationStart();
```

Comment:
Should we move this line down to be the first line in `_validateIndexName`?

Reply:
No, _validateIndexName is debounced and we want the button to be disabled until validation is complete. Putting this.props.onIndexNameValidationStart() in the _validateIndexName would delay calling 
 this.props.onIndexNameValidationStart() until debounce is satisfied which would lose this user feedback.